mlir/lib/Catalyst/Transforms/TBAAPatterns.cpp

```diff
@@ -31,7 +31,13 @@ void setTag(mlir::Type baseType, catalyst::TBAATree *tree, mlir::MLIRContext *ct
 {
     mlir::LLVM::TBAATagAttr tag;
     if (isa<IndexType>(baseType) || isa<IntegerType>(baseType)) {
-        tag = tree->getTag("int");
+        // Index can be used as a pointer.
+        if (isa<IndexType>(baseType) && (isa<LLVM::StoreOp>(newOp) || isa<LLVM::LoadOp>(newOp))) {
```

Comment:
We could use that as a hack, but I think it would be better to check the defining op of the load and store and check if we have a ptrtoint